src/pages/Locations/LocationDetail/LocationDetail.tsx

```diff
@@ -0,0 +1,65 @@
+import type { FC } from 'react';
+import { useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import { IonContent, IonPage, IonSegment, IonSegmentButton, IonLabel, IonText } from '@ionic/react';
+import { CommonPageHeader, Loader } from 'src/components';
+import type { Location } from '../../../types';
+import useLocation from '../hooks/useLocation';
+import LocationHours from '../LocationHours';
+import LocationInfo from '../LocationInfo';
+import '../LocationItem.css';
+
+enum ClassType {
+  Location = 'location',
+  Hours = 'hours',
+}
+
+const LocationDetail: FC = () => {
+  const { t } = useTranslation();
+
+  // @ts-ignore
+  const { locationId } = useParams();
+  const [selectedSegment, setSelectedSegment] = useState<ClassType>(ClassType.Location);
+  const [locationResult, setLocationResult] = useState<Location>();
```

Comment:
this is redundant, you can use useMemo instead of this state and useEffect below.